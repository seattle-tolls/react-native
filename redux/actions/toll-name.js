import { SWITCH_TOLL_NAME } from '../constants/action-types'

export const switchTollName = (payload) => ({
  type: SWITCH_TOLL_NAME, payload,
})