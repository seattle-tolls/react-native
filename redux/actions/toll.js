import { SET_TOLL_INFO, FETCH_TOLL_INFO } from '../constants/action-types'

export const setTollInfo = (payload) => ({
  type: SET_TOLL_INFO, payload,
})

export const fetchTollInfo = () => ({
  type: FETCH_TOLL_INFO,
})