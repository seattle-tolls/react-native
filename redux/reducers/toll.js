import { SET_TOLL_INFO } from '../constants/action-types'
import { createSelector } from 'reselect'

const initialState = {}

const tollReducer = (toll = initialState, action) => {
  switch (action.type) {
    case SET_TOLL_INFO:
      return action.payload
    default:
      return toll
  }
}

export const tollInfo = (state) => {
  const tollInfoObj = state.toll[state.tollName]
  console.log('TOLL INFO', tollInfoObj && tollInfoObj.name)
  return tollInfoObj
}

// FIXME: fix the other selectors according with the derived data

// export const todayInfo = state => state.toll.todayInfo
// export const todaySchedule = state => state.toll.todaySchedule

// export const todayScheduleAsArray = createSelector(
//   todaySchedule,
//   (schedule) => {
//     if(schedule[0]){
//       const scheduleArray = Object.keys(schedule).map(hour => {
//         return ({
//           id: hour,
//           goodToGo: schedule[hour].goodToGo,
//           payByMail: schedule[hour].payByMail,
//           schedule: schedule[hour].schedule,
//         })
//       })
//       return scheduleArray
//     }
//   }
// )

// export const currentTimeInfo = createSelector(
//   todayInfo,
//   todaySchedule,
//   (info, schedule) => {
//     for(let i = info.hour; i >= 0; i--){
//       if(schedule[i]){
//         return{
//           dayName: info.dayName,
//           holidayName: info.holidayName,
//           ...schedule[i],
//         }
//       }
//     }
//   }
// )

export default tollReducer
