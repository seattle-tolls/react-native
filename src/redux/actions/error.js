import { REPORT_ERROR } from '../constants/action-types'

export const reportError = (error) => ({
  type: REPORT_ERROR, error,
})