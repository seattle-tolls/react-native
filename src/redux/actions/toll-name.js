import { SET_TOLL_NAME } from '../constants/action-types'

export const setTollName = (payload) => ({
  type: SET_TOLL_NAME, payload,
})
