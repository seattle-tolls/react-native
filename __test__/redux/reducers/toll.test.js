import tollReducer from '../../../src/redux/reducers/toll'
import { setTollInfo } from '../../../src/redux/actions/toll'

import payload from '../../mock/toll-info-normalized.json'

describe('REDUCERS', () => {
  describe('reducer/toll', () => {
    const initialState = tollReducer('undefined', { type: 'INIT' })
    it('should set the toll reducer', () => {
      const action = setTollInfo(payload)
      const reducer = tollReducer(initialState, action)

      expect(reducer).toEqual(payload)
      expect(reducer['toll99'].name).toBe('99 Tunnel')
      expect(reducer['toll520'].name).toBe('520 Bridge')
      expect(reducer).toMatchSnapshot()
    })

    it('should test for unknown action for tollReducer', () => {
      const reducer = tollReducer(initialState, { type: 'FAKE_ACTION' })
      expect(reducer).toBe(initialState)
      expect(reducer).toMatchSnapshot()
    })
  })
})