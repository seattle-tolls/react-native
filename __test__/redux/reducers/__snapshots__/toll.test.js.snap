// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REDUCERS reducer/toll should set the toll reducer 1`] = `
Object {
  "toll520": Object {
    "data": Object {
      "weekdays": Object {
        "0": Object {
          "goodToGo": 1.25,
          "payByMail": 3.25,
          "schedule": "Midnight to 4:59 a.m.",
        },
        "10": Object {
          "goodToGo": 2.7,
          "payByMail": 4.7,
          "schedule": "10 a.m. to 1:59 p.m.",
        },
        "14": Object {
          "goodToGo": 3.4,
          "payByMail": 5.4,
          "schedule": "2 p.m. to 2:59 p.m.",
        },
        "15": Object {
          "goodToGo": 4.3,
          "payByMail": 6.3,
          "schedule": "3 p.m. to 5:59 p.m.",
        },
        "18": Object {
          "goodToGo": 3.4,
          "payByMail": 5.4,
          "schedule": "6 p.m. to 6:59 p.m.",
        },
        "19": Object {
          "goodToGo": 2.7,
          "payByMail": 4.7,
          "schedule": "7 p.m. to 8:59 p.m.",
        },
        "21": Object {
          "goodToGo": 2,
          "payByMail": 4,
          "schedule": "9 p.m. to 10:59 p.m.",
        },
        "23": Object {
          "goodToGo": 1.25,
          "payByMail": 3.25,
          "schedule": "11 p.m. to 11:59 p.m.",
        },
        "5": Object {
          "goodToGo": 2,
          "payByMail": 4,
          "schedule": "5 a.m. to 5:59 a.m.",
        },
        "6": Object {
          "goodToGo": 3.4,
          "payByMail": 5.4,
          "schedule": "6 a.m. to 6:59 a.m.",
        },
        "7": Object {
          "goodToGo": 4.3,
          "payByMail": 6.3,
          "schedule": "7 a.m. to 8:59 a.m.",
        },
        "9": Object {
          "goodToGo": 3.4,
          "payByMail": 5.4,
          "schedule": "9 a.m. to 9:59 a.m.",
        },
      },
      "weekend": Object {
        "0": Object {
          "goodToGo": 1.25,
          "payByMail": 3.25,
          "schedule": "Midnight to 4:59 a.m.",
        },
        "11": Object {
          "goodToGo": 2.65,
          "payByMail": 4.65,
          "schedule": "11 a.m. to 5:59 p.m.",
        },
        "18": Object {
          "goodToGo": 2.05,
          "payByMail": 4.05,
          "schedule": "6 p.m. to 8:59 p.m.",
        },
        "21": Object {
          "goodToGo": 1.4,
          "payByMail": 3.4,
          "schedule": "9 p.m. to 10:59 p.m.",
        },
        "23": Object {
          "goodToGo": 1.25,
          "payByMail": 3.25,
          "schedule": "11 p.m. to 11:59 p.m.",
        },
        "5": Object {
          "goodToGo": 1.4,
          "payByMail": 3.4,
          "schedule": "5 a.m. to 7:59 a.m.",
        },
        "8": Object {
          "goodToGo": 2.05,
          "payByMail": 4.05,
          "schedule": "8 a.m. to 10:59 a.m.",
        },
      },
    },
    "date": 1577906804692,
    "name": "520 Bridge",
  },
  "toll99": Object {
    "data": Object {
      "weekdays": Object {
        "15": Object {
          "goodToGo": 2.25,
          "payByMail": 4.25,
          "schedule": "3 p.m. to 6 p.m.",
        },
        "18": Object {
          "goodToGo": 1.25,
          "payByMail": 3.25,
          "schedule": "6 p.m. to 11 p.m.",
        },
        "23": Object {
          "goodToGo": 1,
          "payByMail": 3,
          "schedule": "11 p.m. to 6 a.m.",
        },
        "6": Object {
          "goodToGo": 1.25,
          "payByMail": 3.25,
          "schedule": "6 a.m. to 7 a.m.",
        },
        "7": Object {
          "goodToGo": 1.5,
          "payByMail": 3.5,
          "schedule": "7 a.m. to 9 a.m.",
        },
        "9": Object {
          "goodToGo": 1.25,
          "payByMail": 3.25,
          "schedule": "9 a.m. to 3 p.m.",
        },
      },
      "weekend": Object {
        "0": Object {
          "goodToGo": 1,
          "payByMail": 3,
          "schedule": "All day",
        },
      },
    },
    "date": 1577906804692,
    "name": "99 Tunnel",
  },
}
`;

exports[`REDUCERS reducer/toll should test for unknown action for tollReducer 1`] = `"undefined"`;
